### PR TITLE
feat: context module support inline loader

### PR DESCRIPTION
**Pull request by faga295**
## Require Documentation?

- [ ] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

File: webpack-test/cases/loaders/context/test.filter.js [hunk 1]
```diff
@@ -1,12 +0,0 @@
-
-/*
-const supportsRequireInModule = require("../../../helpers/supportsRequireInModule");
-
-module.exports = config => {
```
Comment (bvanjoi) [inline webpack-test/cases/loaders/context/test.filter.js:5]: In my opinion, this ﻿`module.exports` should not be deleted.

File: crates/rspack_core/src/context_module_factory.rs [hunk 2]
```diff
@@ -86,7 +86,12 @@ impl ContextModuleFactory {
     let mut file_dependencies = Default::default();
     let mut missing_dependencies = Default::default();
     let context_dependencies = Default::default();
-    let specifier = dependency.request();
+    let request = dependency.request().to_string();
```
Comment (bvanjoi) [inline crates/rspack_core/src/context_module_factory.rs:89]: `.to_string` is not neccessary.

File: packages/rspack/tests/configCases/loader/context/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+it("should be able to use a context with a loader", () => {
+	let a = "a";
+	let result = require(`raw-loader!./${a}`).default;
```
Comment (bvanjoi) [inline packages/rspack/tests/configCases/loader/context/index.js:3]: There's no requirement to utilise ﻿`raw-loader` in this. I suggest creating a `﻿loader.js` file with the following code:

```js
module.exports = (content) => content;
```

and then replacing ﻿`require(raw-loader!./${a}﻿)` with `﻿require(./loader.js!./${a}﻿)`.

**Review by bvanjoi — COMMENTED** [review]
Once these nits are addressed and the ci pass, I believe it's ready to merge.

File: webpack-test/cases/loaders/context/test.filter.js (hunk 1 above)
Comment (faga295) [reply webpack-test/cases/loaders/context/test.filter.js:5]: I delete test.filter.js according to https://github.com/web-infra-dev/rspack/blob/77f57e52180462019274bada5a2b77e90a6ffca4/webpack-test/README.md?plain=1#L17-L24

File: packages/rspack/tests/configCases/loader/context/index.js (hunk 3 above)
Comment (faga295) [reply packages/rspack/tests/configCases/loader/context/index.js:3]: IMO, using raw-loader can help confirm if the loader is functioning as expected. The `loader.js` file appears to return the original content, making it challenging to determine whether `loader.js` is working.

Comment (h-a-n-a) [reply packages/rspack/tests/configCases/loader/context/index.js:3]: To test if the loader is working, It's not necessary to use an additional loader.
Even a `content.replace("1", "2")` is enough.

Comment (faga295) [reply packages/rspack/tests/configCases/loader/context/index.js:3]: > To test if the loader is working, It's not necessary to use an additional loader. Even a `content.replace("1", "2")` is enough.

get~

**Pull request merged**